MetaMask/metamask-extension
# Add Staking entrypoint for Mainnet on Token List Item

## **Description**

Staking team would like to add an entry point for users to get to Portfolio Staking from the extension.
This PR adds a prop to TokenListItem, which would add the staking button for a given token when the flag is true.
The scope of the story is limited to ETH mainnet, with the expectation that there will be other tokens to stake in the future.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1781

## **Manual testing steps**

1. Go to Token Listing 
2. Verify that the new staking icon shows only for ETH mainnet
3. Verify that onClick, a new metametrics event is tracked

## **Screenshots/Recordings**



### **Before**



### **After**


<img width="355" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/10986371/e99dbb87-33ef-49b7-b0dd-79ae8570d12e">
<img width="359" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/10986371/5e3d04a1-0c00-4cbc-9737-b87f26168147">

## **Pre-merge author checklist**

- [] I’ve included tests if applicable

## **Pre-merge reviewer checklist**
